content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,342 @@
+can add such parameters as a list of `steps` whose logs we are interested in
+or `page` and `size` parameters to paginate the logs output if it gets too
+verbose. For example, let's say that we want to quickly check the last 15 lines
+of logs of the second `fitdata` step of our workflow. We can do that as follows:
+
+```python
+import json
+from reana_client.api.client import get_workflow_logs
+
+workflow_logs = get_workflow_logs(my_workflow_name, my_reana_token, ['fitdata'])
+job_logs = json.loads(workflow_logs['logs'])['job_logs']
+fitdata_logs = job_logs[next(iter(job_logs))]['logs']
+
+for item in fitdata_logs.split('\n')[-15:]:
+    print(item)
+```
+
+Output:
+
+```python
+ERR MATRIX APPROXIMATE
+ PARAMETER  CORRELATION COEFFICIENTS
+       NO.  GLOBAL      1      2      3      4      5
+        1  0.00000   1.000  0.000  0.000  0.000  0.000
+        2  0.00000   0.000  1.000  0.000  0.000  0.000
+        3  0.00000   0.000  0.000  1.000  0.000  0.000
+        4  0.00000   0.000  0.000  0.000  1.000  0.000
+        5  0.00000   0.000  0.000  0.000  0.000  1.000
+ ERR MATRIX APPROXIMATE
+[#1] INFO:Minization -- RooMinimizer::optimizeConst: deactivating const optimization
+[#1] INFO:Plotting -- RooAbsPdf::plotOn(model) directly selected PDF components: (bkg)
+[#1] INFO:Plotting -- RooAbsPdf::plotOn(model) indirectly selected PDF components: ()
+Info in <TCanvas::Print>: png file results/plot.png has been created
+```
+
+### Listing workspace files
+
+Once the workflow is finished, we usually want to list all the output files which our workflow
+produced. With REANA client it can be achieved by using
+[ls](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-ls)
+command. Similarly, it can be done in the following manner:
+
+```python
+from reana_client.api.client import list_files
+
+list_files(my_workflow_name, my_reana_token)
+```
+
+Output:
+
+```python
+[
+  {
+    'last-modified': '2021-02-24T09:57:01',
+    'name': 'code/gendata.C',
+    'size': 1937,
+  },
+  {
+    'last-modified': '2021-02-24T09:57:01',
+    'name': 'code/fitdata.C',
+    'size': 1648,
+  },
+  {
+    'last-modified': '2021-02-24T09:57:52',
+    'name': 'results/plot.png',
+    'size': 15450,
+  },
+  {
+    'last-modified': '2021-02-24T09:57:42',
+    'name': 'results/data.root',
+    'size': 154458,
+  },
+]
+```
+
+### Downloading outputs
+
+After listing all the files we may want to download some of the results for inspection.
+With REANA client we could use [download](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-download)
+command. In this case to download a binary blob of a final plot which our
+workflow produced all we need to do is to use `download_file` function:
+
+```python
+from reana_client.api.client import download_file
+
+output_filename = 'results/plot.png'
+file_binary_blob = download_file(
+    my_workflow_name, output_filename, my_reana_token)
+```
+
+### Displaying output images
```

Comment:
We can now display the final fit produced by our workflow:


...

Example output:

...